aws/lambda-with-other-services/send-sms-code-with-sns/send-sms-code-with-sns.md

```diff
@@ -0,0 +1,123 @@
+
+```javascript
+ SNS.publish(params).promise();
+ ```
+
+ This is what we will be returning at the end of our lambda function.
+
+ Next, we can set up a function that will generate a verification code to send to our user once they input their information as shown below.
+
+ ```javascript
+    //...
+    	let smsCode = '';
+
+	//this function will generate a random number with d amount of digits. zeros can be the first digit.
+	function generate(d) {
+
+		let max = Math.pow(10, d)*9-1;
+		let min = Math.pow(10, d);
+
+		let number = Math.floor( Math.random() * max + min);
+		smsCode = ('' + number).substring(1);
+	}
+
+	generate(6);
+    //...
+```
+
+ To do this, first set a variable called smsCode to an empty string inside of our exports.handler function but outside of our new function. This variable will get redefined by our generate function and can be used inside our params down below to be sent to the user.
+
+ We want to generate a code with 6 digits and have them all be numeric characters. The parameter "d" is the amount of digits we want to generate. At the bottom, you can see we are generating a 6 digit code when we run the function.
+
+ The variable "max" is a formula that essentially boils down to the number 8,999,999 when the parameter is set to 6. The variable "min" boils down to 1,000,000. The amount of digits specified will always result in a min and max of that amount of digits plus one. This will help us later on.
+
+After those are set up, we will create a formula to generate a random number shown below:
+
+```javascript
+let number = Math.floor( Math.random() * max + min);
+```
+
+Here we are multipling Math.random by our max. In this case, that would equal anywhere from 0 to 8,999,999. Then we add on our "min" variable, which will bring up our range to anywhere from 1,000,000 to 9,999,999. This is where our 7 digit number will help us out. As you can see, the last 6 digits of any number in that range is anywhere from 000,000 to 999,999. For this reason, we do not care what number is in the millions place. The 7th digit is only there so our 6 digit number can begin with a zero, therefore giving us a truely random 6 digit number. To finish, we redefine the variable "smsCode" by turning our 7 digit number into a string and then use the substring(1) function to only include the 2nd digit onwards. (e.g. The number 4092134 will return the string '092134') This code is shown below:
+
+```javascript
+smsCode = ('' + number).substring(1);
+```
+
+When we add all the pieces together for this verification code text message, we get the following code:
+
+```javascript
+const AWS = require('aws-sdk');
+
+exports.handler = (event, context, cb) => {
+	const SNS = new AWS.SNS();
+
+	let smsCode = '';
+
+	//this function will generate a random number with d amount of digits. zeros can be the first digit.
+	function generate(d) {
```

Comment:
It's not considered good practice to declare a function within the handler. However, you can declare it outside the scope of the handler and still call it in here.